p2p/src/net/types/mod.rs

```diff
@@ -46,7 +46,7 @@ pub struct AddrInfo<T: NetworkingService> {
 /// (both are Mintlayer nodes and that both support mandatory protocols). If those checks pass,
 /// the information is passed on to [crate::peer_manager::PeerManager] which decides whether it
 /// wants to keep the connection open or close it and possibly ban the peer from.
-#[derive(Debug)]
+#[derive(Debug, Eq)]
```

Comment:
I think this is playing with fire. We shouldn't inherit Eq here. What does it mean that PeerInfo A == PeerInfo B? What if peer_id of A == peer_id of B, but the agent is different. Does that mean they're not the same peer?

Please understand that I know that this is "impossible" based on the invariants of our software. But you gotta ask yourself what it means that PeerInfo can have equality, but not the peer_id. I think this is wrong and should be removed, and wherever equality is needed should be examined and we should understand "what we're trying to prove" with that equality, and then see what equality we should do there. Maybe just equating the peer_id is more than enough. In the case of trait-bounds, maybe we should create another trait that satisfies this property.

I'm a huge opponent of throwing Eq and other inherited traits without carefully considering what that means. It can easily become a bug.

Reply:
`Eq` is only needed in one test (`connect_to_remote`). I think it will be better rewire the test instead.